### PR TITLE
fix: UNIFY-1230 handle errors when loading version information

### PR DESCRIPTION
**Pull request by ryanthemanuel**
- Closes UNIFY-1230

### User facing changelog

Fixes determining whether there is an update available when there is an error loading version information or metadata

### Additional details

I am mimicking what we do today in version 9. Specifically we do not indicate that there is a new version available in this scenario:

https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/updater.js#L55-L56

### PR Tasks

- [X] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/graphql/schemas/schema.graphql [hunk 1]
```diff
@@ -742,7 +742,7 @@ type Mutation {
   cloudProjectCreate(ciProviders: [String!], name: String!, orgId: ID!, public: Boolean!): CloudProject
 
   """Request access to an organization from a projectId"""
-  cloudProjectRequestAccess(projectSlug: String!): Boolean
```
Comment (tgriesser) [inline packages/graphql/schemas/schema.graphql:745]: It looks like this branch might be out of date w/ 10.0-release, looks like it already has this change: https://github.com/cypress-io/cypress/blob/567034459089d9d53dfab5556cb9369fb335c3db/packages/graphql/schemas/schema.graphql#L910-L911

**Review by tgriesser — COMMENTED** [review]
Nice - what do you think about adding an open mode test which checks that we can get through the launchpad flow with the network disconnected (faking `ctx.fetch` with whatever error we'd typically see when the app is offline). Or we could do that as a separate pr

**Review by tgriesser — CHANGES_REQUESTED** [review]
Added a note for how we can avoid the `// @ts-ignore`

File: packages/app/cypress/e2e/top-nav.cy.ts [hunk 2]
```diff
@@ -205,6 +207,35 @@ describe('App Top Nav Workflows', () => {
         cy.findAllByRole('dialog').should('not.exist')
       })
     })
+
+    context('version data unreachable', () => {
+      it('treats unreachable data as current version', () => {
+        cy.withCtx((ctx) => {
+          const oldFetch = ctx.util.fetch
+
+          // @ts-ignore sinon is a global in the node process where this is executed
```
[the remark this answers is not in the document]
Comment (emilyrohrbough) [reply packages/app/cypress/e2e/top-nav.cy.ts:216]: Could you also do [`Cypress.sinon`](https://docs.cypress.io/api/utilities/sinon#Syntax) to remove the type error

File: packages/app/cypress/e2e/top-nav.cy.ts [hunk 3]
```diff
@@ -205,6 +207,35 @@ describe('App Top Nav Workflows', () => {
...
+          sinon.stub(ctx.util, 'fetch').get(() => {
```
Comment (emilyrohrbough) [inline packages/app/cypress/e2e/top-nav.cy.ts:217]: would you be able to use a spy on fetch and verify it wasn't called with either url?

Comment (ryanthemanuel) [reply packages/app/cypress/e2e/top-nav.cy.ts]: @emilyrohrbough I actually am expecting these to be called with those urls and want to throw an error when they are.

File: packages/app/cypress/e2e/top-nav.cy.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ryanthemanuel) [reply packages/app/cypress/e2e/top-nav.cy.ts]: @tgriesser this is now fixed.

**Review by emilyrohrbough — APPROVED** [review]

**Review by tgriesser — APPROVED** [review]
Nice!

**Pull request merged**